cmd/system-probe/api/server.go

```diff
@@ -33,10 +33,10 @@ func StartServer(cfg *config.Config) error {
 	}
 
 	// Register stats endpoint
-	mux.HandleFunc("/debug/stats", func(w http.ResponseWriter, req *http.Request) {
+	mux.HandleFunc("/debug/stats", utils.WithConcurrencyLimit(5, func(w http.ResponseWriter, req *http.Request) {
```

Comment:
At minimum, a const at the top instead of magic number?
Also, should this have a configurable?

Reply:
Good point – I added a const and a comment

Reply:
looks like you didn't use the const here though.